### PR TITLE
Repository service.

### PR DESCRIPTION
**Pull request by materiywakal**
Repository mapper implemented.
We can now save repository to database.

### CODE REVIEW

File: NugetAnalyzer.BLL/Interfaces/IRepositoryMapper.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+using NugetAnalyzer.Domain;
+
+namespace NugetAnalyzer.BLL.Interfaces
```
Comment (pryabov) [inline NugetAnalyzer.BLL/Interfaces/IRepositoryMapper.cs]: @gennadybaranov what do you think. For me we need to get rid from this interface and class at all

**Review by pryabov — CHANGES_REQUESTED** [review]

File: NugetAnalyzer.BLL/Interfaces/IRepositoryMapper.cs (hunk 1 above)
Comment (materiywakal) [reply NugetAnalyzer.BLL/Interfaces/IRepositoryMapper.cs]: Maybe we don't need this interface at all and we don't need this class as dependency but we can use it as separate entity which we can initialize and use in methods. Also i can do this class disposable so we can include using construction.

Comment (materiywakal) [reply NugetAnalyzer.BLL/Interfaces/IRepositoryMapper.cs]: Bad thoughts coming at 4 a.m.

Comment (pryabov) [reply NugetAnalyzer.BLL/Interfaces/IRepositoryMapper.cs]: :) If you need to discuss something, pls contact with me directly

File: NugetAnalyzer.BLL/Services/RepositoryService.cs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            packageRepository = unitOfWork.GetRepository<Package>();
+            packageVersionRepository = unitOfWork.GetRepository<PackageVersion>();
+        }
+
+        public async void SaveAsync(Models.Repositories.Repository repository, int userId)
+        {
+            unitOfWork.GetRepository<Repository>().Add(await ToDomainAsync(repository, userId));
+            unitOfWork.SaveChangesAsync();
+        }
+
```
Comment (pryabov) [inline NugetAnalyzer.BLL/Services/RepositoryService.cs:67]: Add code region to the file
`#region Private Methods`
And when in some other part of BLL will be needed the same functionality method can be extracted to some kind of helper

File: NugetAnalyzer.BLL/Services/RepositoryService.cs [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            {
+                databaseRepository.Delete(dbRepository.Id);
+            }
+
+            var domainRepository = CreateRepository(businessRepository.Name, userId);
+
+            foreach (var solution in businessRepository.Solutions)
+            {
+                domainRepository.Solutions.Add(CreateSolution(solution.Name));
+                var domainSolution = domainRepository.Solutions.FirstOrDefault(o => o.Name == solution.Name);
```
Comment (pryabov) [inline NugetAnalyzer.BLL/Services/RepositoryService.cs]: `o` - pl, use more sensible name

File: NugetAnalyzer.BLL/Services/RepositoryService.cs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        {
+            return new Project
+            {
+                Name = name,
+                ProjectPackageVersions = new List<ProjectPackageVersion>()
+            };
+        }
+
+        private async Task<ProjectPackageVersion> CreatePackageAsync(
+            string packageName, string version, Project domainProject, Repository domainRepository)
```
Comment (pryabov) [inline NugetAnalyzer.BLL/Services/RepositoryService.cs]: in this case each argument from new line
```
private async Task<ProjectPackageVersion> CreatePackageAsync(
            string packageName, 
string version, 
Project domainProject, 
Repository domainRepository)
```

**Review by pryabov — CHANGES_REQUESTED** [review]

**Review by pryabov — APPROVED** [review]

**Pull request merged**